src/_tutorials/language/futures.md

```diff
@@ -367,17 +382,31 @@ written in a couple different ways.
 
   <?code-excerpt "misc/lib/tutorial/daily_news.dart (main-future-api-pass-print)"?>
   {% prettify dart %}
-  printDailyNewsDigest() =>
+  Future<void> printDailyNewsDigest() =>
       gatherNewsReports().then(print);
   {% endprettify %}
 
+You need to provide an argument to `then()`'s callback,
```

Comment:
I like that you're covering this because its counter-intuitive. (or, wait, did I write this? anyways! :))

I wonder if there's a way to present this as more of a simple mechanism, rather than as cautionary advice.

"the method `then()` on `Future` takes a callback with a single parameter. That parameter's type will then be inferred, so its a `String` on a `Future<String>`. The same mechanism occurs for `Future<void>`, so you get a parameter of type `void`, which means the compiler won't let you accidentally use it."

OK not nearly as concise, and, probably more confusing.

I wonder if we can eventually support things like:

```
  StreamController<void>().add(); // simplification of .add(null); since the parameter is void
  Future<void>().then(() { ... }); // don't require a parameter if its void
```

That would be slick.